starlite/openapi/controller.py

```diff
@@ -109,12 +154,25 @@ def retrieve_schema_json(self, request: Request) -> "OpenAPI":
                 A [Request][starlite.connection.Request] instance.
 
         Returns:
-            A rendered JSON object..
+            resoponse: With a rendered YAML object
         """
-        return self.get_schema_from_request(request)
+        if not request.app.openapi_config:  # pragma: no cover
+            raise ImproperlyConfiguredException("Starlite has not been instantiated with OpenAPIConfig")
+
+        if self.should_serve_endpoint(request):
+            return Response(
+                content=self.get_schema_from_request(request),
+                status_code=HTTP_200_OK,
+                media_type=OpenAPIMediaType.OPENAPI_JSON,
+            )
+        return Response(
```

Comment:
same comment basically as above regarding this response not being of the correct type. This should be a `OpenAPIMediaType.OPENAPI_JSON` response with an empty dictionary.